driver-core/src/main/java/com/datastax/driver/core/DateWithoutTime.java

```diff
@@ -0,0 +1,260 @@
+ * <ul>
+ *     <li>it uses a proleptic gregorian calendar, meaning that it's gregorian indefinitely back in the past (there is no gregorian change);</li>
+ *     <li>there is a year 0.</li>
+ * </ul>
+ *
+ * This class implements these differences, so that year/month/day fields match exactly the ones in
+ * CQL string literals.
+ *
+ * @since 2.2
+ */
+public final class DateWithoutTime {
+
+    private static final TimeZone UTC = TimeZone.getTimeZone("UTC");
+
+    private final long millisSinceEpoch;
+    private final int daysSinceEpoch;
+
+    // This gets initialized lazily if we ever need it. Once set, it is effectively immutable.
+    private volatile GregorianCalendar calendar;
+
+    private DateWithoutTime(int daysSinceEpoch) {
+        this.daysSinceEpoch = daysSinceEpoch;
+        this.millisSinceEpoch = TimeUnit.DAYS.toMillis(daysSinceEpoch);
+    }
+
+    /**
+     * Builds a new instance from a number of days since January 1st, 1970 GMT.
+     *
+     * @param daysSinceEpoch the number of days.
+     *
+     * @return the new instance.
+     */
+    public static DateWithoutTime fromDaysSinceEpoch(int daysSinceEpoch) {
+        return new DateWithoutTime(daysSinceEpoch);
+    }
+
+    /**
+     * Builds a new instance from a number of milliseconds since January 1st, 1970 GMT.
+     *
+     * @param millisSinceEpoch the number of milliseconds. Note that if it does not correspond to a whole number of days,
+     *                         it will be rounded towards 0.
+     *
+     * @return the new instance.
+     *
+     * @throws IllegalArgumentException if the date is not in the range [-5877641-06-23; 5881580-07-11].
+     */
+    public static DateWithoutTime fromMillisSinceEpoch(long millisSinceEpoch) throws IllegalArgumentException {
+        long daysSinceEpoch = TimeUnit.MILLISECONDS.toDays(millisSinceEpoch);
+        checkArgument(daysSinceEpoch >= Integer.MIN_VALUE && daysSinceEpoch <= Integer.MAX_VALUE,
+            "Date should be in the range [-5877641-06-23; 5881580-07-11]");
+
+        return new DateWithoutTime((int)daysSinceEpoch);
+    }
+
+    /**
+     * Builds a new instance from a year/month/day specification.
+     *
+     * Note that, for internal implementation reasons, the month and day are not lenient
+     * (in fact, the day is still lenient if <= 31, but this class makes no guarantee
+     * that this will remain true in the future).
+     *
+     * @param year the year in ISO format ({@link DateWithoutTime this class's Javadoc}).
+     * @param month the month. It is 1-based (e.g. 1 for January) and not lenient.
+     * @param dayOfMonth the day of the month.
+     *
+     * @return the new instance.
+     *
+     * @throws IllegalArgumentException if the date is not in the range [-5877641-06-23; 5881580-07-11],
+     * or the month is not between 1 and 12, or the day is not between 1 and 31.
+     */
+    public static DateWithoutTime fromYearMonthDay(int year, int month, int dayOfMonth) {
```

Comment:
> (in fact, the day is still lenient if <= 31, but this class makes no guarantee that this will remain true in the future)

Not allowing > 31 is necessary for the year hack to work, otherwise the user could provide (0,12,32) which is really (1,1,1), but we would shift the year by one because it's <= 0.

I saw no value in making the underlying calendar lenient, but in hindsight that's more confusing than useful, so I will follow Robert's suggestion.